src/components/Home/Home.js

```diff
@@ -17,49 +20,98 @@ function Home () {
   useEffect(() => {
     async function position () {
       try {
-        await navigator.geolocation.getCurrentPosition(function (position) {
-          setLatitude(position.coords.latitude)
-          setLongitude(position.coords.longitude)
-        })
+        const now = new Date()
+        if (JSON.parse(localStorage.getItem('location') === null) || now.getTime() > JSON.parse(localStorage.getItem('location')).expiryTime) {
+          await navigator.geolocation.getCurrentPosition((pos) => {
+            if (pos.coords.latitude && pos.coords.longitude) {
+              setLatitude(pos.coords.latitude)
+              setLongitude(pos.coords.longitude)
```

Comment:
Rather than separately changing latitude and then longitude. Can we make one reducer for location state?